workitem/workitem_repository_blackbox_test.go

```diff
@@ -533,3 +536,151 @@ func (s *workItemRepoBlackBoxTest) TestConcurrentWorkItemCreations() {
 		assert.Equal(s.T(), 0, report.failures)
 	}
 }
+
+func (s *workItemRepoBlackBoxTest) TestList() {
+	s.T().Run("list with explicit order", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.Iterations(1),
+			tf.WorkItems(10, func(fxt *tf.TestFixture, idx int) error {
+				switch idx {
+				case 0, 1, 2, 3, 4, 5, 6:
+					fxt.WorkItems[idx].Fields[workitem.SystemState] = "open"
+				default:
+					fxt.WorkItems[idx].Fields[workitem.SystemState] = "new"
+				}
+				return nil
+			}),
+		)
+		t.Run("by created descending", func(t *testing.T) {
+			// when
+			exp, _ := query.Parse(ptr.String(`{"system.state": "open"}`))
+			sort, _ := workitem.ParseSortWorkItemsBy(ptr.String("-created"))
+			res, count, err := s.repo.List(context.Background(), fxt.Spaces[0].ID, exp, nil, nil, nil, sort)
+			// then
+			require.NoError(t, err)
+			assert.Equal(t, 7, count)
+			toBeFound := id.Slice{
+				fxt.WorkItems[0].ID,
+				fxt.WorkItems[1].ID,
+				fxt.WorkItems[2].ID,
+				fxt.WorkItems[3].ID,
+				fxt.WorkItems[4].ID,
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+			}.ToMap()
+			for i := 0; i <= 6; i++ {
+				require.Equal(t, fxt.WorkItems[i].ID, res[6-i].ID)
+			}
+			for _, wi := range res {
+				_, ok := toBeFound[wi.ID]
+				require.True(t, ok, "unknown work item found: %s", wi.ID)
+				delete(toBeFound, wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to found all work items: %+s", toBeFound)
+		})
+		t.Run("by created ascending", func(t *testing.T) {
+			// when
+			exp, _ := query.Parse(ptr.String(`{"system.state": "open"}`))
+			sort, _ := workitem.ParseSortWorkItemsBy(ptr.String("created"))
+			res, count, err := s.repo.List(context.Background(), fxt.Spaces[0].ID, exp, nil, nil, nil, sort)
+			// then
+			require.NoError(t, err)
+			assert.Equal(t, 7, count)
+			toBeFound := id.Slice{
+				fxt.WorkItems[0].ID,
+				fxt.WorkItems[1].ID,
+				fxt.WorkItems[2].ID,
+				fxt.WorkItems[3].ID,
+				fxt.WorkItems[4].ID,
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+			}.ToMap()
+			for i := 0; i <= 6; i++ {
+				require.Equal(t, fxt.WorkItems[i].ID, res[i].ID)
+			}
+			for _, wi := range res {
+				_, ok := toBeFound[wi.ID]
+				require.True(t, ok, "unknown work item found: %s", wi.ID)
+				delete(toBeFound, wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to found all work items: %+s", toBeFound)
+		})
+		t.Run("by updated descending", func(t *testing.T) {
+			// when
+			s.repo.Save(context.Background(), fxt.WorkItems[0].SpaceID, *fxt.WorkItems[3], fxt.Identities[0].ID)
```

Comment:
@baijum I know all the workitems share the same SpaceID but wouldnt it be better if you use actual workitem's spaceID for each Save() 

like

`s.repo.Save(context.Background(), fxt.WorkItems[3].SpaceID, *fxt.WorkItems[3], fxt.Identities[0].ID)`
`s.repo.Save(context.Background(), fxt.WorkItems[3].SpaceID, *fxt.WorkItems[2], fxt.Identities[0].ID)`

Reply:
Fixed in fc4532d72dbae3555f995aec00b4266cfc524cf2